examples/encoding/plot_rerp_encoder.py

```diff
@@ -0,0 +1,67 @@
+"""
+========================================
+Regression on continuous event-related data (rER[P/F])
+========================================
+
+This demonstrates how rER[P/F]s - regressing the continuous data - is a
+generalisation of traditional averaging. If all preprocessing steps
+are the same, no overlap between epochs exists, and all predictors are binary,
+regression is virtually identical to traditional averaging.
+
+If overlap exists and/or predictors are continuous, traditional averaging
+is inapplicable, but regression can still estimate effects.
+
+rERPs are described in:
+
+    Smith, N. J., & Kutas, M. (2015). Regression-based estimation of ERP
+    waveforms: II. Non-linear effects, overlap correction, and practical
+    considerations. Psychophysiology, 52(2), 169-189.
+"""
+# Authors: Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Chris Holdgraf <choldgraf@gmail.com>
+#
+# License: BSD (3-clause)
+
+import matplotlib.pyplot as plt
+
+import mne
+from mne.datasets import sample
+from mne.encoding.model import EventRelatedRegressor
+from mne.encoding.feature import clean_inputs
+
+# Load and preprocess data
+data_path = sample.data_path()
+raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+raw = mne.io.read_raw_fif(raw_fname, preload=True).pick_types(
+    meg='grad', stim=True, eeg=False).filter(1, None, method='iir')
+
+# Set up events
+events = mne.find_events(raw)
+event_id = {'Aud/L': 1, 'Aud/R': 2}
+tmin, tmax = -.1, .5
+
+# Regular epoching
+picks = mne.pick_types(raw.info, meg=True)
+epochs = mne.Epochs(raw, events, event_id, tmin, tmax, reject=None,
+                    baseline=None, preload=True, verbose=False)
+
+# rERF
+rerp = EventRelatedRegressor(raw, events, est='cholesky', event_id=event_id,
+                             tmin=tmin, tmax=tmax,
+                             preproc_func_xy=clean_inputs)
```

Comment:
Well initially `EncodingModel` was going to take Raw arrays etc, but it was suggested to instead allow `EncodingModel` to take numpy arrays instead, and then to build higher-level objects on top that did more specific things. I think this is one of the points we should discuss in meeting.

>  Yes - though for maximal convenience, this would not have to be user addressed. Ie., dealt with as here.
 
That's why I created the `EventRelatedRegressor` object - so that if people didn't want to, they didn't have to deal with creating their own processing pipelines and instead could use a "stock" pipeline to do standard analyses (e.g., like an rER[P/F]). It sounds like I made `EventRelatedRegressor` not flexible enough (e.g., not being able to take continuous data), so maybe it's worth changing it a bit, and maybe renaming the class, so that it can handle both continual and events-based data (plus diff time windows for different event types etc)